### ISSUE TITLE
Build errors out due to usage of "-Wno-static-in-inline" option

### ISSUE THREAD

**Issue by george-b** [neovim/neovim#5300]
- `nvim --version`: N/A
- Vim (version: ) behaves differently? N/A
- Operating system/version: SLES 11 SP3 (Linux)
- Terminal name/version:  tmux
- `$TERM`: screen-256color

### Actual behaviour

Build errors out due to usage of "-Wno-static-in-inline" option.

### Expected behaviour

Build to complete without error.

### Steps to reproduce using `nvim -u NORC`

N/A

I experience the following build error.

```
ninja  -C .deps
ninja: Entering directory `.deps'
[2/2] cd /users/gbrown/src/binstrap/src/neovim/.deps && /users/gbrown/local/cmake/bin/cmake -E touch .third-party
ninja  -C build
ninja: Entering directory `build'
[10/105] Building C object src/nvim/CMakeFiles/nvim.dir/eval.c.o
FAILED: /usr/bin/cc  -DHAVE_CONFIG_H -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -Iconfig -I../src -isystem ../.deps/usr/include -Isrc/nvim/auto -Iinclude -Wconversion -O2 -g  -DDISABLE_LOG   -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector --param ssp-buffer-size=4  -Wno-static-in-inline -Wno-conversion -MD -MT src/nvim/CMakeFiles/nvim.dir/eval.c.o -MF src/nvim/CMakeFiles/nvim.dir/eval.c.o.d -o src/nvim/CMakeFiles/nvim.dir/eval.c.o   -c ../src/nvim/eval.c
cc1: error: unrecognized command line option "-Wno-static-in-inline"
[10/105] Building C object src/nvim/CMakeFiles/nvim.dir/eval/encode.c.o
ninja: build stopped: subcommand failed.
make: *** [nvim] Error 1
```

This is from a pull from master.

```
git rev-parse HEAD
3cccee2a82e36fcf0843012bf12c7095bd21af03
```

Platform details are as follows.

```
gcc --version
gcc (SUSE Linux) 4.3.4 [gcc-4_3-branch revision 152973]
Copyright (C) 2008 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

lsb_release -a
LSB Version:    core-2.0-noarch:core-3.2-noarch:core-4.0-noarch:core-2.0-x86_64:core-3.2-x86_64:core-4.0-x86_64:desktop-4.0-amd64:desktop-4.0-noarch:graphics-2.0-amd64:graphics-2.0-noarch:graphics-3.2-amd64:graphics-3.2-noarch:graphics-4.0-amd64:graphics-4.0-noarch
Distributor ID: SUSE LINUX
Description:    SUSE Linux Enterprise Server 11 (x86_64)
Release:        11
Codename:       n/a

cat /etc/SuSE-release
SUSE Linux Enterprise Server 11 (x86_64)
VERSION = 11
PATCHLEVEL = 3

uname -a
Linux cflcs01 3.0.101-0.47.71-default #1 SMP Thu Nov 12 12:22:22 UTC 2015 (b5b212e) x86_64 x86_64 x86_64 GNU/Linux
```

And for completeness I have the following in `local.mk`

```
cat local.mk
CMAKE_EXTRA_FLAGS="-DCMAKE_INSTALL_PREFIX:PATH=/users/gbrown/local/neovim"
CMAKE_BUILD_TYPE := RelWithDebInfo
```

I am aware this is an old and clunky platform but I have been able to build neovim for some time without issue so I'm not sure if this is intentional (not sure which version of GCC introduces "-Wno-static-in-inline")? To be clear I totally understand if the maintainers don't want to support builds on such a platform.

I will try to bisect at which commit this became an issue later but an initial review of the recent commits doesn't show anything obvious to me.

**Comment by george-b** [comment]
> I will try to bisect at which commit this became an issue later but an initial review of the recent commits doesn't show anything obvious to me.

Ah the bisect shows this is 7e2348f2b1b487c875bbcf6c6711a276f9063040 I must have misread `git blame` as this is the only place I see this in the source code.

**Comment by bfredl** [comment]
Recent enough compilers seem to realize that `-Wno-` could be safely ignored (at least, warning not error) if not present, as the warning is not present then anyway.... 

gcc 4.4 introduced this behavior according to https://gcc.gnu.org/gcc-4.4/changes.html, maybe we should add a guard for <= gcc  4.3

**Comment by george-b** [comment]
> gcc 4.4 introduced this behavior according to https://gcc.gnu.org/gcc-4.4/changes.html, maybe we should add a guard for <= gcc 4.3

Thanks for tracking this down! Looks like RHEL 6.8 has gcc 4.4.7, TBH I'd consider that a reasonable cut-off. Unless others hit this I'm happy for this to be closed, I doubt anyone else is building on such an out of date system.

**Comment by fwalch** [comment]
If we don't add a guard, we should update the [build prerequisites](https://github.com/neovim/neovim/wiki/Building-Neovim#build-prerequisites).

**Comment by bfredl** [comment]
Done. If more interest is expressed in compiling with 4.3 we could change it back, but as gcc 4.4 was released in 2009...

**Comment by george-b** [comment]
As the build prerequisites I'll go ahead and close this.

**Comment by equalsraf** [comment]
The default gcc shipped in OpenBSD is 4.2, a pkg provides 4.9. Updating the wiki instructions.
